components/Tabs/src/TabBarView/MDCTabBarView.m

```diff
@@ -91,6 +99,74 @@ - (void)setSelectedItem:(UITabBarItem *)selectedItem {
   }
 
   _selectedItem = selectedItem;
+  UIView *itemView = self.itemViews[itemIndex];
+  if ([itemView isKindOfClass:[MDCTabBarViewItemView class]]) {
+    MDCTabBarViewItemView *selectedItemView = (MDCTabBarViewItemView *)itemView;
+    selectedItemView.image = selectedItem.selectedImage ?: selectedItem.image;
+  }
+}
+
+#pragma mark - Key-Value Observing (KVO)
+
+- (void)addObserversToTabBarItems {
+  for (UITabBarItem *item in self.items) {
+    [item addObserver:self
+           forKeyPath:kImageString
+              options:NSKeyValueObservingOptionNew
+              context:nil];
+    [item addObserver:self
+           forKeyPath:kSelectedImageString
+              options:NSKeyValueObservingOptionNew
+              context:nil];
+    [item addObserver:self
+           forKeyPath:kTitleString
+              options:NSKeyValueObservingOptionNew
+              context:nil];
+  }
+}
+
+- (void)removeObserversFromTabBarItems {
+  for (UITabBarItem *item in self.items) {
+    @try {
+      [item removeObserver:self forKeyPath:kImageString];
+      [item removeObserver:self forKeyPath:kSelectedImageString];
+      [item removeObserver:self forKeyPath:kTitleString];
+    } @catch (NSException *exception) {
+      if (exception) {
```

Comment:
Is this if statement necessary?

Reply:
No, I can pass the crash along to clients. I prefer it, honestly, but our team tends not to support that type of debugging efforts.

Reply:
Done.